File: library/src/main/java/cz/ackee/rxoauth/DefaultErrorChecker.java
package cz.ackee.rxoauth;

import java.net.HttpURLConnection;

import retrofit2.HttpException;

/**
 * Default errro checker that checks for 401 for expired access token and 401 + 400 for bad refresh of token
 *
 * @author David Bilik [david.bilik@ackee.cz]
 * @since 02/12/2017
 **/
public class DefaultErrorChecker implements ErrorChecker {
    @Override
    public boolean isExpiredAccessToken(Throwable t) {
        if (t instanceof HttpException) {
            if (((HttpException) t).code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
                return true;
            }
        }
        return false;
    }

    @Override
    public boolean isBadRefreshToken(Throwable t) {
        if (t instanceof HttpException) {
            if (((HttpException) t).code() == HttpURLConnection.HTTP_BAD_REQUEST || ((HttpException) t).code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
                return true;
            }
        }
        return false;
    }
}

File: core/src/main/java/cz/ackee/rxoauth/DefaultErrorChecker.kt
package cz.ackee.rxoauth

import retrofit2.HttpException
import java.net.HttpURLConnection

/**
 * Default error checker that checks for 401 for expired access token and 401 + 400 for bad refresh of token
 */
class DefaultErrorChecker : ErrorChecker {

    override fun invalidAccessToken(t: Throwable): Boolean {
        if (t is HttpException) {
            if (t.code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
                return true
            }
        }
        return false
    }

    override fun invalidRefreshToken(t: Throwable): Boolean {
        if (t is HttpException) {
            if (t.code() == HttpURLConnection.HTTP_BAD_REQUEST || t.code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
                return true
            }
        }
        return false
    }
}
